core/chains/evm/config/toml/config.go

```diff
@@ -311,13 +314,26 @@ func (c *EVMConfig) ValidateConfig() (err error) {
 		err = multierr.Append(err, commonconfig.ErrMissing{Name: "Nodes", Msg: "must have at least one node"})
 	} else {
 		var hasPrimary bool
+		var logBroadcasterEnabled bool
+		if c.LogBroadcasterEnabled != nil {
+			logBroadcasterEnabled = *c.LogBroadcasterEnabled
+		}
+
 		for _, n := range c.Nodes {
 			if n.SendOnly != nil && *n.SendOnly {
 				continue
 			}
-			hasPrimary = true
-			break
+
+			if !hasPrimary {
```

Comment:
nit: if statement is redundant as it's fine to set `hasPrimary = true` if it's already true